ruby/tests/common_tests.rb

```diff
@@ -0,0 +1,1111 @@
+                                      :optional_enum => :C,
+                                      :repeated_string => ["hello", "there", "world"])
+    assert m.optional_int32 == -42
+    assert m.optional_msg.class == proto_module::TestMessage2
+    assert m.repeated_string.length == 3
+    assert m.optional_enum == :C
+    assert m.repeated_string[0] == "hello"
+    assert m.repeated_string[1] == "there"
+    assert m.repeated_string[2] == "world"
+  end
+
+  def test_ctor_string_symbol_args
+    m = proto_module::TestMessage.new(:optional_enum => 'C', :repeated_enum => ['A', 'B'])
+    assert_equal :C, m.optional_enum
+    assert_equal [:A, :B], m.repeated_enum
+
+    m = proto_module::TestMessage.new(:optional_string => :foo, :repeated_string => [:foo, :bar])
+    assert_equal 'foo', m.optional_string
+    assert_equal ['foo', 'bar'], m.repeated_string
+  end
+
+  def test_ctor_nil_args
+    m = proto_module::TestMessage.new(:optional_enum => nil, :optional_int32 => nil, :optional_string => nil, :optional_msg => nil)
+
+    assert_equal :Default, m.optional_enum
+    assert_equal 0, m.optional_int32
+    assert_equal "", m.optional_string
+    assert_nil m.optional_msg
+  end
+
+  def test_embeddedmsg_hash_init
+    m = proto_module::TestEmbeddedMessageParent.new(:child_msg => {sub_child: {optional_int32: 1}},
+                                                    :number => 2,
+                                                    :repeated_msg => [{sub_child: {optional_int32: 3}}],
+                                                    :repeated_number => [10, 20, 30])
+
+    assert_equal 2, m.number
+    assert_equal [10, 20, 30], m.repeated_number
+
+    assert_not_nil m.child_msg
+    assert_not_nil m.child_msg.sub_child
+    assert_equal m.child_msg.sub_child.optional_int32, 1
+
+    assert_not_nil m.repeated_msg
+    assert_equal 1, m.repeated_msg.length
+    assert_equal 3, m.repeated_msg.first.sub_child.optional_int32
+  end
+
+  def test_inspect
+    m = proto_module::TestMessage.new(:optional_int32 => -42,
+                                      :optional_enum => :A,
+                                      :optional_msg => proto_module::TestMessage2.new,
+                                      :repeated_string => ["hello", "there", "world"])
+    expected = "<#{proto_module}::TestMessage: optional_int32: -42, optional_int64: 0, optional_uint32: 0, optional_uint64: 0, optional_bool: false, optional_float: 0.0, optional_double: 0.0, optional_string: \"\", optional_bytes: \"\", optional_msg: <#{proto_module}::TestMessage2: foo: 0>, optional_enum: :A, repeated_int32: [], repeated_int64: [], repeated_uint32: [], repeated_uint64: [], repeated_bool: [], repeated_float: [], repeated_double: [], repeated_string: [\"hello\", \"there\", \"world\"], repeated_bytes: [], repeated_msg: [], repeated_enum: []>"
+    assert_equal expected, m.inspect
+  end
+
+  def test_hash
+    m1 = proto_module::TestMessage.new(:optional_int32 => 42)
```

Comment:
map? oneof?

Reply:
Ignore me if this PR doesn't deal with to_h.

Reply:
Sorry, this test is for hash values. The test for `to_h` exist in the respective test files for proto2 and proto3 (`basic_proto2.rb` and `basic.rb`)